controllers/machine.go

```diff
@@ -527,6 +539,104 @@ func byHardwareAffinity(hardware []tinkv1.Hardware, preferred []infrastructurev1
 	}, nil
 }
 
+// ensureHardwareProvisionJob ensures the hardware is ready to be provisioned.
+// Uses the BMCRef from the hardware to create a BMCJob.
+// The BMCJob is responsible for getting the machine to desired state for provisioning.
+// If a BMCJob is already found and has failed, we error.
+func (mrc *machineReconcileContext) ensureHardwareProvisionJob(hardware *tinkv1.Hardware) error {
```

Comment:
curious why the use of `HardwareProvisionJob` instead of `BMCJob`? This requires a translation that feels like it might be unnecessary.

Reply:
I intended it to be like "Job that gets hardware ready for provisioning" since it does multiple actions. 
I tried to explain it a bit on the function comment.

Reply:
The challenge i see is that a `HardwareProvisionJob` is just a pointer to a BMCJob. The definition you give in the comments points us to a BMCJob. So to understand the result or behavior of this function we need to go BMCJob regardless. I know naming can be very subjective and difficult but I'm having trouble seeing the value in creating this new construct of a `HardwareProvisionJob`.